packages/botonic-nlu/src/botonic-nlu.js

```diff
@@ -0,0 +1,184 @@
+import path from 'path'
+import { readJSON, readDir } from './file-utils'
+import { detectLang, preprocessData } from './preprocessing'
+import { getEmbeddingMatrix } from './word-embeddings'
+import * as tf from '@tensorflow/tfjs-node'
+import { parseLangFlag, printPrettyConfig } from './utils'
+import {
+  UTTERANCES_DIRNAME,
+  MODELS_DIRNAME,
+  NLU_DATA_FILENAME,
+  MODEL_FILENAME
+} from './constants'
+import { loadDevData, saveDevData } from './file-utils'
+import { getPrediction, getIntent } from './prediction'
+import { getEntities } from './ner'
+
+// TODO: interactive command to try intents from terminal
+// import inquirer from 'inquirer'
+// import { interactiveMode } from './scripts/interactive-mode'
+// async function askForInteractiveMode() {
+//   const questions = [
+//     {
+//       type: 'confirm',
+//       name: 'affirmative',
+//       message: `Do you want to switch into interactive mode?`
+//     }
+//   ]
+//   return inquirer.prompt(questions)
+// }
+
+export class BotonicNLU {
+  constructor() {
+    this.languages = parseLangFlag(process.argv)
+    this.nluPath = ''
+    this.utterancesPath = ''
+    this.modelsPath = ''
+    this.devData = {}
+    this.models = {}
+  }
+
+  async train({ nluPath }) {
+    // TODO: Think about passing an arg for using models in memory
+    this.nluPath = nluPath
+    this.utterancesPath = path.join(nluPath, UTTERANCES_DIRNAME)
+    this.modelsPath = path.join(nluPath, MODELS_DIRNAME)
+    this.devData = loadDevData(this.nluPath, this.languages)
+    this.languages = Object.keys(this.devData)
+    for (let language of this.languages) {
+      let devData = this.devData[`${language}`]
+      let { devIntents, params, devEntities } = devData
+      params = { ...params, LANG: language } // TODO: Think better this reassignment
```

Comment:
what about params.LANG= language ?
Why LANG in capitals?

Reply:
renamed! this was from the old implementation